src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -255,4 +282,195 @@ private static Stream<Arguments> invalidNames() {
         Arguments.of("Te*st")
     );
   }
+
+  @Nested
+  class RequestRoleChange {
+    @Test
+    public void valid()
+        throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+        AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+        AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+        MFARequiredButNotConfiguredException, SQLException {
+      LoginResult currentlyStudentLogin =
+          loginAs(httpSession, ITConstants.TEST_STUDENT_EMAIL, ITConstants.TEST_STUDENT_PASSWORD);
+      HttpServletRequest upgradeRequest = createRequestWithCookies(new Cookie[] {currentlyStudentLogin.cookie});
+      replay(upgradeRequest);
+      Map<String, String> requestDetails = Map.of(
+          "verificationDetails", "school staff url",
+          "otherDetails", "more information"
+      );
+      RegisteredUserDTO initialUserState = currentlyStudentLogin.user;
+      assertFalse(initialUserState.getTeacherPending());
+
+      Response upgradeResponse = usersFacade.requestRoleChange(upgradeRequest, requestDetails);
+
+      assertEquals(Response.Status.OK.getStatusCode(), upgradeResponse.getStatus());
+      assertTrue(upgradeResponse.readEntity(RegisteredUserDTO.class).getTeacherPending());
+
+      // Reset flag
```

Comment:
The reason why you have to do this is because your tests are not independent, which is not a good practice. The result, outcome or execution of one test should not influence other tests. Isn't there any other way around this?

Reply:
The integration tests build the database (and elasticsearch and some other stuff) as a BeforeAll. In order to prevent interference from database contents, we'd either have to split tests in separate files/classes or we'd have to change it to rebuild the database for every test. The former doesn't seem like a good idea and given the performance on the integration tests as-is, I'm not sure increasing the frequency of container population is viable either...